doc/weakly_supervised.rst

```diff
@@ -317,6 +335,20 @@ implicit assumptions of MMC is that all classes form a compact set, i.e.,
 follow a unimodal distribution, which restricts the possible use-cases of this
 method. However, it is one of the earliest and a still often cited technique.
 
+This is the first Mahalanobis distance learning method, the algorithm aims at 
+maximizing the sum of distances between all the instances from the dissimilar 
```

Comment:
> Yes, I agree that more doc for additional algorithms can be added in a next PR. It would be nice to be able to merge this soon. @hansen7 do you expect to have time to address the small comments?

Hi, @wdevazelhes @bellet , thanks for the strong support! I have resolved the listed the issues, and add the mathematical descriptions of all the metric learning algorithm but the `cov`. Please see attached, for the latex part, it is all work locally, and I have standardise the notations:

   1. uppercase bolded letters for matrices 
   2. lowercase bolded letters for vectors
   3. uppercase unbolded letters for sets
   4. lowercase unbolded letters for scalars